compiler/hash-typecheck/src/traverse/mod.rs

```diff
@@ -101,11 +110,22 @@ impl<'gs, 'ls, 'cd, 'src> visitor::AstVisitor for TcVisitor<'gs, 'ls, 'cd, 'src>
         node: hash_ast::ast::AstNodeRef<hash_ast::ast::AccessName>,
     ) -> Result<Self::AccessNameRet, Self::Error> {
         // Accumulate all the names into an access term:
-        let builder = self.builder();
         let mut names = node.path.iter();
-        let mut current_term = builder.create_var_term(*names.next().unwrap().body());
+        let source_id = self.source_id;
+
+        let first_name = names.next().unwrap();
+        let location = self.source_location(first_name.span());
+
+        let builder = self.builder();
+        let mut current_term = builder.create_var_term(*first_name.body());
+
+        builder.add_location_to(current_term, location);
+
         for access_name in names {
-            current_term = builder.create_ns_access(current_term, *access_name.body())
+            let name_location = SourceLocation { span: access_name.span(), source_id };
```

Comment:
Can use `self.source_location` here.

Reply:
You cannot use `self.source_location` since the builder is being instantiated above and that would mean that it's being borrowed as mutable and the immutable...